app/src/test/java/io/intrepid/contest/screens/join/JoinPresenterTest.java

```diff
@@ -58,23 +62,39 @@ public void onEntryCodeTextViewChangedShouldShowSubmitButtonWhenCodeIsNotEmpty()
     }
 
     @Test
-    public void onSubmitButtonClickedShouldShowEntryNameScreenWhenCodeIsValid() {
+    public void onSubmitButtonClickedShouldShowEntryNameScreenWhenCodeIsValidForContestant() {
```

Comment:
Yes, but that function is basically hardcoded to skip to next screen. I have extracted it to another method to clarify that's what it does.
I have tested it for the coverage, but that test doesn't mean anything (I would not manually `setParticipationType`, that'd come from the API).